Mathlib/CategoryTheory/Generator/Presheaf.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import Mathlib.CategoryTheory.Generator.Coproduct
+import Mathlib.CategoryTheory.Limits.FunctorCategory.Basic
+
+/-!
+# Generators in the category of presheaves
+
+In this file, we show that if `A` is a preadditive category that has a separator (and
+suitable coproducts), then the category of presheaves `Cᵒᵖ ⥤ A` also has a separator.
+
```

Comment:
The category of presheaves `Cᵒᵖ ⥤ Type` also has a separator. Is there a reason you can't prove it in a generality which includes this important case too?

Reply:
Actually, I show the result for presheaves under the assumption `A` has zero morphisms (for sheaves, we do not have `HasZeroMorphisms` instances, which is the reason why I have assumed `A` is preadditive).
For presheaves in `Type`, the construction gives a generating *set* of objects (the Yoneda presheaves), but I do not know how to pass from this to a separator unless we have zero morphisms. (Actually, it seems to me that there is no separator in general, e.g. when `C := Discrete (Fin 2)`.)

Reply:
Oops, you're right - I misremembered.